flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/KnnModelData.java

```diff
@@ -0,0 +1,165 @@
+                    Tuple3.of(
+                            (DenseMatrix) row.getField(0),
+                            (DenseVector) row.getField(1),
+                            (int[]) row.getField(2)));
+        }
+        comparator = Comparator.comparingDouble(o -> -o.f0);
+    }
+
+    /**
+     * Gets comparator.
+     *
+     * @return Comparator.
+     */
+    public Comparator<? super Tuple2<Double, Integer>> getQueueComparator() {
+        return comparator;
+    }
+
+    /**
+     * Gets dictionary data size.
+     *
+     * @return Dictionary data size.
+     */
+    public Integer getLength() {
+        return dictData.size();
+    }
+
+    public List<Tuple3<DenseMatrix, DenseVector, int[]>> getDictData() {
+        return dictData;
+    }
+
+    /** Encoder for the Knn model data. */
+    public static class ModelDataEncoder implements Encoder<Row> {
+        @Override
+        public void encode(Row modelData, OutputStream outputStream) throws IOException {
+            DataOutputView dataOutputView = new DataOutputViewStreamWrapper(outputStream);
+
+            DenseMatrixSerializer matrixSerializer = new DenseMatrixSerializer();
+            matrixSerializer.serialize((DenseMatrix) modelData.getField(0), dataOutputView);
+
+            DenseVectorSerializer vectorSerializer = new DenseVectorSerializer();
+            vectorSerializer.serialize((DenseVector) modelData.getField(1), dataOutputView);
+
+            int[] label = (int[]) Objects.requireNonNull(modelData.getField(2));
+            for (Integer integer : label) {
+                dataOutputView.writeInt(integer);
+            }
+        }
+    }
+
+    /** Decoder for the Knn model data. */
+    public static class ModelDataStreamFormat extends SimpleStreamFormat<Row> {
+
+        @Override
+        public Reader<Row> createReader(Configuration config, FSDataInputStream stream) {
+            return new Reader<Row>() {
+
+                @Override
+                public Row read() throws IOException {
+                    try {
+                        DataInputView source = new DataInputViewStreamWrapper(stream);
+                        DenseMatrix matrix = new DenseMatrixSerializer().deserialize(source);
+                        DenseVector vector = new DenseVectorSerializer().deserialize(source);
+                        int[] label = new int[vector.size()];
+                        for (int i = 0; i < label.length; ++i) {
+                            label[i] = source.readInt();
+                        }
+                        return Row.of(matrix, vector, label);
+                    } catch (EOFException e) {
+                        return null;
+                    }
+                }
+
+                @Override
+                public void close() throws IOException {
+                    stream.close();
+                }
+            };
+        }
+
+        @Override
+        public TypeInformation<Row> getProducedType() {
```

Comment:
`KMeansModelData` and `LogisticRegressionModelData` do not override `getProducedType`. Could we do the same here for simplicity?

Reply:
![image](https://user-images.githubusercontent.com/40415963/145357394-d0ca57b2-8b3c-4146-b69d-3e622f222d98.png)

KMeans and LR also override this function, but the return value are different.

Reply:
OK, I think I catch your idea. I will refine it later.